pkg/runtime/tags.go

```diff
@@ -23,19 +22,59 @@ import (
 	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
 )
 
+// expandTagFormat is a function which returns the resolved value for an
+// ACK resource tag format. Ex: %CONTROLLER_SERVICE% -> s3
+type expandTagFormat func(rtclient.Object, acktypes.ServiceControllerMetadata) string
```

Comment:
What is `rtclient.Object` used for?

Reply:
If this function calls `rtclient.Object` for future ResourceTagFormat we should also return an `error`

Reply:
`rtClient.Object` is used to retrieve `Name` and `Namespace` fields from the Object metadata.

I chose to stay away from error handling when resolving Tag formats, and instead use the pattern of injecting default values if the field is missing. Ex: `MissingImageTagValue` for the controller-version field.